code/game/machinery/computer/_computer.dm

```diff
@@ -22,7 +22,7 @@
 	if(!QDELETED(C))
 		qdel(circuit)
 		circuit = C
-		C.loc = null
```

Comment:
if it's already qdeleted it shouldn't need to be moved to nullspace this doesn't make any sense

Reply:
if C ISN'T qdeleted, qdel circuit and make C also point to circuit, move circuit to nullspace. It's not moving a thing that is deleted.